hs_core/management/commands/migrate_swat_model_instance_resources.py

```diff
@@ -0,0 +1,317 @@
+                        data['modelObjective'][obj_fld] = True
+                        break
+            if other_objectives:
+                data['modelObjective']['otherObjectives'] = other_objectives
+
+        if 'modelParameter' in data:
+            model_parameter = data['modelParameter']
+            model_parameters = model_parameter['model_parameters']
+            other_parameters = '' if model_parameter['other_parameters'] is None else \
+                model_parameter['other_parameters'].strip()
+
+            data['modelParameter'] = {"fertilizer": False, "pointSource": False, "cropRotation": False,
+                                      "tileDrainage": False, "tillageOperation": False, "irrigationOperation": False,
+                                      "inletOfDrainingWatershed": False}
+
+            param_to_json_field_map = {"cropRotation": "Crop rotation", "tileDrainage": "Tile drainage",
+                                       "pointSource": "Point source", "fertilizer": "Fertilizer",
+                                       "tillageOperation": "Tillage operation",
+                                       "inletOfDrainingWatershed": "Inlet of draining watershed",
+                                       "irrigationOperation": "Irrigation operation"}
+            for param_fld in data['modelParameter']:
+                for param in model_parameters:
+                    if param['description'] == param_to_json_field_map[param_fld]:
+                        data['modelParameter'][param_fld] = True
+                        break
+            if other_parameters:
+                data['modelParameter']['otherParameters'] = other_parameters
+
+        if 'modelMethod' in data:
+            model_method_fields = ('flowRoutingMethod', 'petEstimationMethod', 'runoffCalculationMethod')
+            model_method = data['modelMethod']
+            for fld in model_method_fields:
+                if model_method[fld] is None or not model_method[fld].strip():
+                    model_method.pop(fld)
+
+        if 'simulationType' in data:
+            simulation_type_name = '' if data['simulationType']['simulation_type_name'] is None \
+                else data['simulationType']['simulation_type_name'].strip()
+            if simulation_type_name:
+                data['simulationType'] = simulation_type_name
+            else:
+                data.pop('simulationType')
+
+        return data
+
+    def handle(self, *args, **options):
+        logger = logging.getLogger(__name__)
+        resource_counter = 0
+        err_resource_counter = 0
+        to_resource_type = 'CompositeResource'
+        swat_resource_count = SWATModelInstanceResource.objects.count()
+        msg = "THERE ARE CURRENTLY {} SWAT MODEL INSTANCE RESOURCES TO MIGRATE TO COMPOSITE RESOURCE.".format(
+            swat_resource_count)
+        logger.info(msg)
+        self.stdout.write(self.style.SUCCESS(msg))
```

Comment:
Why both logging and stout?  I usually see print in management commands.

Reply:
Logging the migration operations to keep a reference in case we find issues with a migrated resource. If you think logging may not be useful I can remove.  

Using stdout instead of print to make the message to bit standout with different color for error message vs success message when running the command from the command line - was helpful when developing and testing this command. This of course is of not much use when this command is executed as a Jenkins job.